### PR TITLE
Fix `outLength` returned by `dataWithOSCBlobBytes:maxLength:length:`

### PR DESCRIPTION
**Pull request by balord**
Parsing an OSC message can fail when message data follows OSC Blob data.

When reading a buffer of OSC message data, Blob data is converted using NSData category method `-[NSData dataWithOSCBlobBytes:maxLength:length:]`. F53OSCParser then advances the cursor according to the `outLength` returned by that method. Currently, that length is not guaranteed to be a multiple of 4 bytes. This can cause subsequent OSC types in the same OSC message to be parsed incorrectly because the buffer is no longer aligned at a 32-bit boundary.

This PR updates the NSData+F53OSCBlob category method `-[NSData dataWithOSCBlobBytes:maxLength:length:]` to follow the same pattern the related NSString+F53OSCString category method uses. In addition, this PR also guards against some potential NULL dereferences in the handling of `outLength`.

### CODE REVIEW

File: NSData+F53OSCBlob.m [hunk 1]
```diff
@@ -51,22 +51,44 @@ - (NSData *) oscBlobData
...
     if ( buf == NULL || maxLength == 0 )
+    {
+        if ( outLength != NULL )
+            *outLength = 0;
         return nil;
+    }
     
     UInt32 dataSize = 0;
     
     dataSize = *((UInt32 *)buf);
```
Comment (ChrisAshworth) [inline NSData+F53OSCBlob.m]: I noticed that in `F53OSCNumber` we cast the unswapped int32 to an `SInt32` before calling `OSSwapBigToHostInt32`.  At first blush it seems to me that it probably doesn't matter which one we're using since we immediately swap it to host order, but maybe I'm wrong, or if I'm not wrong, is it better to be consistent? I don't have a strong opinion here. Maybe better to just leave it since we know it works?

File: F53OSCParser.m [hunk 2]
```diff
@@ -188,6 +188,7 @@ + (nullable F53OSCMessage *) parseOscMessageData:(NSData *)data
                 switch ( type )
                 {
                     case 's':
+                        dataLength = 0; // reset
```
[the remark this answers is not in the document]
Comment (balord) [reply F53OSCParser.m]: Yes, it is correct to say it's redundant. It's just my small preference to be assured of the reset inline in F53OSCParser without needing to jump over to NSString+F53OSCString.

[the remark this answers is not in the document]
Comment (ChrisAshworth) [reply F53OSCParser.m]: 👍

File: F53OSCParser.m [hunk 3]
```diff
@@ -205,12 +206,13 @@ + (nullable F53OSCMessage *) parseOscMessageData:(NSData *)data
...
                         break;
                     case 'b':
+                        dataLength = 0; // reset
                         dataArg = [NSData dataWithOSCBlobBytes:buffer maxLength:lengthOfRemainingBuffer length:&dataLength];
                         if ( dataArg != nil )
                         {
                             [args addObject:dataArg];
-                            buffer += dataLength + 4;
-                            lengthOfRemainingBuffer -= (dataLength + 4);
+                            buffer += dataLength;
```
[the remark this answers is not in the document]
Comment (balord) [reply F53OSCParser.m]: yes - great idea

File: NSData+F53OSCBlob.m (hunk 1 above)
Comment (balord) [reply NSData+F53OSCBlob.m]: I would be in favor of leaving this as-is.
We could always add some tests to verify edge cases are handled correctly.
